client/src/main/java/io/pravega/client/stream/impl/EventStreamReaderImpl.java

```diff
@@ -110,7 +113,8 @@
     }
     
     private EventRead<Type> readNextEventInternal(long timeout) throws ReaderNotInReaderGroupException, TruncatedDataException {
-        long waitTime = Math.min(timeout, ReaderGroupStateManager.TIME_UNIT.toMillis());
+        int numReaders = Math.max(1, groupState.getNumberOfReaders());
```

Comment:
`groupState.getNumberOfReaders()` cannot return less 1.

Reply:
Thanks, this was a kind of defensive setting. If your statement is correct, then maybe I can just use the result of `groupState.getNumberOfReaders()`.

Reply:
I don't really understand why the number of readers should effect this calculation.